### PR TITLE
Adding Go API view generator

### PR DESCRIPTION
**Pull request by catalinaperalta**
This PR adds the token generator for Go API views.

### CODE REVIEW

**Comment by catalinaperalta** [comment]
NOTE: the output is still a WIP, params are currently still not evenly outputted across different Go types

File: src/go/cmd/content.go [hunk 1]
```diff
@@ -54,6 +54,10 @@ func (c *content) addConst(pkg pkg, g *ast.GenDecl) {
 				// const FooConst = FooType("value")
 				co.Type = pkg.getText(ce.Fun.Pos(), ce.Fun.End())
 				v = pkg.getText(ce.Args[0].Pos(), ce.Args[0].End())
+			} else if ce, ok := vs.Values[0].(*ast.BinaryExpr); ok {
```
Comment (jhendrixMSFT) [inline src/go/cmd/content.go:57]: Add a test to cover this scenario?

File: src/go/cmd/testdata/test_struct/test.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package teststruct
```
Comment (jhendrixMSFT) [inline src/go/cmd/testdata/test_struct/test.go:4]: Add internal struct that doesn't show up?

File: src/go/cmd/token_builders.go [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+// Licensed under the MIT License.
+
+package cmd
+
+import (
+	"strings"
+)
+
+func addTab() string {
+	return "     "
```
Comment (jhendrixMSFT) [inline src/go/cmd/token_builders.go:11]: Should this return "\t" instead?

File: src/go/cmd/token_builders.go [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+}
+
+func makeToken(defID, navID *string, val string, kind int, list *[]Token) {
```
Comment (jhendrixMSFT) [inline src/go/cmd/token_builders.go:14]: It was not clear from invocations below that this mutates state.  Typical pattern is to mimic the semantics of `append()`,  So something like this.
```go
func appendToken(list []Token, defID, navID *string, val string, kind int) []Token { //...}
```
Probably not worth refactoring at this point but something to keep in mind for the future.

Also, comments explaining how this is used would be useful as there's a mix of seemingly optional params (that should really be at the end) and required(?) params.  Maybe a variadic options pattern would work better here?

File: src/go/cmd/api_view.go [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+)
+
+const (
+	text          = 0
```
Comment (jhendrixMSFT) [inline src/go/cmd/api_view.go:16]: Use strongly-typed const here.

```go
type TokenType int

const (
    text TokenType  = 0
)
```

File: src/go/cmd/api_view.go [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+	lineIDMarker  = 5
+	typeName      = 4
+	memberName    = 7
+	stringLiteral = 8
+	literal       = 9
+	comment       = 10
+)
+
+var (
+	reservedNames = map[string]bool{
```
Comment (jhendrixMSFT) [inline src/go/cmd/api_view.go:30]: I'm guessing `bool` is not really important here, i.e. you just wanted a hash set?  Pattern in Go is to use `struct{}` for the value type instead.

File: src/go/cmd/content.go (hunk 1 above)
Comment (catalinaperalta) [reply src/go/cmd/content.go:57]: Done

File: src/go/cmd/testdata/test_struct/test.go (hunk 2 above)
Comment (catalinaperalta) [reply src/go/cmd/testdata/test_struct/test.go:4]: Done

File: src/go/cmd/token_builders.go (hunk 3 above)
Comment (catalinaperalta) [reply src/go/cmd/token_builders.go]: Done and removed addTab() since it does seem unnecessary

File: src/go/cmd/token_builders.go (hunk 4 above)
Comment (catalinaperalta) [reply src/go/cmd/token_builders.go]: Added a comment above makeToken for a possible future refactoring to improve the function signature

File: src/go/cmd/api_view.go (hunk 5 above)
Comment (catalinaperalta) [reply src/go/cmd/api_view.go]: Done

File: src/go/cmd/api_view.go (hunk 6 above)
Comment (catalinaperalta) [reply src/go/cmd/api_view.go]: Done

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by catalinaperalta** [comment]
Had to issue override for check-enforcer since there are no other applicable checks.

**Pull request merged**